lib/cisco_node_utils/dns_domain.rb

```diff
@@ -0,0 +1,79 @@
+#
+# NXAPI implementation of DnsDomain class
+#
+# September 2015, Hunter Haugen
+#
+# Copyright (c) 2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# "group" is a standard SNMP term but in NXOS "role" is used to serve the
+# purpose of group; thus this provider utility does not create snmp groups
+# and is limited to reporting group (role) existence only.
+
+require File.join(File.dirname(__FILE__), 'node_util')
+
+module Cisco
+  # DnsDomain - node utility class for DNS search domain config management
+  class DnsDomain < NodeUtil
+    attr_reader :name, :vrf
+
+    def initialize(name, vrf=nil, instantiate=true)
+      unless name.is_a? String
+        fail TypeError, "Expected a string, got a #{name.class.inspect}"
+      end
+      @name = name
+      @vrf = vrf
+      create if instantiate
+    end
+
+    def self.dnsdomains(vrf=nil)
+      if vrf.nil?
+        domains = config_get('dnsclient', 'domain')
+      else
+        domains = config_get('dnsclient', 'domain_vrf', vrf)
+      end
+      return {} if domains.nil?
+
+      hash = {}
+      domains.each do |name|
+        hash[name] = DnsDomain.new(name, vrf, false)
```

Comment:
@glennmatthews 
> With regard to #23, I would prefer `hash[vrf][name]` over `hash["${vrf}:${name}"]` - that's the style used in most of our other types/providers and avoids the possibility of any issues with special characters. `vrf` will just be `'default'` when no vrf is specified.

I've instead gone a more simple route and just stuck with `hash[name]` as the vrf is required to enumerate the domains, and any API querying for the domains can do a hash-merge between all vrfs as they choose.

There is currently no node-utils library for getting a list of all the vrfs to iterate over, but I don't particularly need it yet. Is that okay?